src/components/search-box.jsx

```diff
@@ -0,0 +1,55 @@
+import React, {PropTypes} from "react"
+
+
+const SearchBox = React.createClass({
+  propTypes: {
+    labelFilter: PropTypes.string,
+    nameFilter: PropTypes.string,
+    onLabelChange: PropTypes.func,
+    onNameChange: PropTypes.func,
+  },
+  onLabelInputChange(event) {
+    this.props.onLabelChange(event.target.value)
+  },
+  onNameInputChange(event) {
+    this.props.onNameChange(event.target.value)
```

Comment:
Hmm that's an interesting discussion (or a troll ^^) but I'm not gonna explain that here.

I'd just say that:
- React is a view library and does not provide any architecture excepted callbacks for events, which are passed to the parent "component"
- Flux architecture has been proposed and its most widespread implementation is the Redux library
- Redux is not used in Legislation Explorer
- `<troll>React+Redux+Webpack+Babel+Ramda+Typescript+... => Elm language + Elm architecture + Elm toolchain</troll>` and I'm now happy with it, and I see through all the comments you do here that you'd like it!